core/trino-parser/pom.xml

```diff
@@ -42,6 +42,12 @@
             <artifactId>antlr4-runtime</artifactId>
         </dependency>
 
+        <dependency>
+            <groupId>io.airlift</groupId>
+            <artifactId>junit-extensions</artifactId>
```

Comment:
This is very brittle. It only works if a module depends on junit-extensions. To ensure it applies to everything, the extension should be registered via the surefire plugin configuration. 


https://github.com/trinodb/trino/pull/18691#discussion_r1297420761

Reply:
There’s a bug or limitation preventing that right now, but yes, that’s the way it should be. 

The benefit of this being in a separate library is that it can be used in *every* module in Trino. The implementation in that other PR limited it to downstream modules due to potential circular deos.